packages/table/src/Table.tsx

```diff
@@ -1,144 +1,195 @@
 import { TableSort } from './types/TableSort';
-import { Cell, ExtendedTableHeader, IdType, Row, TableInstance } from './types/ReactTable';
+import { Cell, Column, IdType, Row, TableInstance, TableOptions, RowProps } from './types/ReactTable';
 import { OnTableClickEvent } from './types/OnTableClickEvent';
 import { OnRowSelectedEvent } from './types/OnRowSelectedEvent';
-import { useTableContext } from './TableContext';
+import { TableContext } from './TableContext';
+import IndeterminateCheckbox from './IndeterminateCheckbox';
+import { TableSortOption } from './types';
+import TableContent from './TableContent';
 
 type HeaderProps = {
   sticky: boolean;
 } & React.HTMLAttributes<HTMLElement>;
 
-export type Props<T extends IdType> = {
+export type CommonTableProps<T extends IdType> = {
   id?: string;
+  tableProps?: React.HTMLAttributes<HTMLElement>;
   bodyProps?: React.HTMLAttributes<HTMLElement>;
-  cellProps?: React.HTMLAttributes<HTMLElement>;
+  getCellProps?: (cell: Cell<T>) => React.HTMLAttributes<HTMLTableCellElement>;
+  getRowProps?: (row: Row<T>) => RowProps;
+
   onCellClick?: (event: OnTableClickEvent<HTMLElement, T>) => void;
   onRowClick?: (event: OnTableClickEvent<HTMLElement, T>) => void;
-  onRowSelected?: (event: OnRowSelectedEvent) => void;
+  onRowSelected?: (event: OnRowSelectedEvent<T>) => void;
   headerProps?: HeaderProps;
-  rowProps?: React.HTMLAttributes<HTMLElement>;
-  scrollable?: boolean;
   onSort?: (sortBy: TableSort[]) => void;
-} & React.HTMLAttributes<HTMLElement>;
+
+  additionalContent?: React.ElementType;
+  additionalContentProps?: Record<string, string | number | boolean | undefined | null>;
+
+  scrollable?: boolean;
+  selectable?: boolean;
+  getCanSelectRow?: (record: T) => boolean;
+  sortable?: boolean;
+  paged?: boolean;
+
+  pluginHooks?: PluginHook<T>[];
+  children?: React.ReactNode | React.ReactChild;
+};
+
+export type TableProps<T extends IdType> = {
+  columns: Column<T>[];
+  data: T[];
+} & React.HTMLAttributes<HTMLElement> &
+  CommonTableProps<T> &
+  TableOptions<T>;
 
 const Table = <T extends IdType>({
+  additionalContent: AdditionalContent,
+  additionalContentProps,
+  columns,
+  data,
+  selectable,
+  scrollable,
+  sortable,
   id,
+  tableProps,
   bodyProps,
-  cellProps,
   headerProps,
   onRowClick,
   onCellClick,
   onRowSelected,
-  rowProps,
-  scrollable,
+  getCanSelectRow,
+  getRowProps = () => ({} as RowProps),
+  getCellProps = () => ({} as React.HTMLAttributes<HTMLTableCellElement>),
   onSort,
+  paged = false,
+  pluginHooks,
+  children,
   ...rest
-}: Props<T>): JSX.Element => {
-  const { sortable, selectable, instance, setScrollable } = useTableContext();
-
-  const {
-    getTableProps,
-    getTableBodyProps,
-    headerGroups,
-    rows,
-    prepareRow,
-    selectedFlatRows: selectedRows,
-    toggleHideColumn,
-  } = instance as TableInstance<T>;
+}: TableProps<T>): JSX.Element | null => {
+  let selectionColumn: Column<T>;
```

Comment:
Why not create the variable inline on 110? Was there some issue with defining it inside the function?

Reply:
Just a bit left from the refactoring. this was used outside of the hook before.